src/cb.h

```diff
@@ -16,9 +16,10 @@ typedef struct circular_buffer
 void cb_init(circular_buffer *cb, size_t capacity)
 {
     cb->buffer = malloc(capacity);
-    if(cb->buffer == NULL)
-        // handle error
-    cb->buffer_end = (char *)cb->buffer + capacity;
+    if(cb->buffer == NULL) {
+        // TODO: handle error
+    }
```

Comment:
the parenthesis is the real error here...

Reply:
Good eye, that will do it. What I get for copy paste and really looking ;)